### PR TITLE
Fix incorrect Javadoc for SortedSetIterable.intersect(),SetIterable.intersect() .

### PR DESCRIPTION
**Pull request by lzcyx**
For issue #1666

### CODE REVIEW

**Comment by motlin** [comment]
Rather than change the Javadoc, I think we should fix the behavior.

**Comment by mohrezaei** [comment]
Just as point of reference, the difference in performance can be drastic if the set sizes are very different. E.g. if one set has 1M elements and the other has 10, changing the code will make this example 100,000x slower.

File: eclipse-collections-code-generator/src/main/resources/api/set/immutablePrimitiveSet.stg [hunk 1]
```diff
@@ -61,7 +61,11 @@ public interface Immutable<name>Set extends Immutable<name>Collection, <name>Set
 
     /**
      * Returns the set of all objects that are members of both {@code this} and {@code set}. The intersection of
-     * [1, 2, 3] and [2, 3, 4] is the set [2, 3].
+     * [1, 2, 3] and [2, 3, 4] is the set [2, 3]. The output will contain instances from the smaller of the two sets
```
Comment (mohrezaei) [inline eclipse-collections-code-generator/src/main/resources/api/set/immutablePrimitiveSet.stg:64]: primitives only have one sense of equality. They are value types, so there is no "memory reference" distinction. Ditto for the other 2 primitive stg files.

Comment (lzcyx) [reply eclipse-collections-code-generator/src/main/resources/api/set/immutablePrimitiveSet.stg]: I apologize for the oversight in my previous commit. I have now reverted these 3 primitive stg files.

**Comment by lzcyx** [comment]
Hi @mohrezaei ,
I wanted to check in on this PR. I've made the requested changes and would appreciate it if you could review them when you have a chance.

Thank you for your time and guidance!

**Comment by mohrezaei** [comment]
LGTM, but I was hoping to get a second opinion from @motlin or @donraab

**Review by motlin — APPROVED** [review]

**Comment by motlin** [comment]
LGTM too, I'll merge it.

**Comment by 0x78f1935** [comment]
I wouldn't trust this dude @lzcyx . Nasty cookie stealer. Wannabe hacker lol

**Pull request merged**